### PR TITLE
chore: remove platform specific, native commandBar styles

### LINKED ISSUE #1418: [Material] Remove "native" CommandBar styles in favor of XAML-based styles
Currently, the Material styles for `CommandBar` exist for a specific use case that has now been deprecated in favor of the `NavigationBar` from Uno Toolkit.

We should remove the current Material `CommandBar` styles and create a single new style for all platforms that simply uses the Material brushes, fonts, corner radii, etc.

### LINKED ISSUE #1425: MaterialToolkitTheme ColorOverrideSource breaks CommandBar content
### Current behavior

If in App.xaml there is
`
<utum:MaterialToolkitTheme
  ColorOverrideSource="ms-appx:///Styles/ColorPaletteOverride.xaml">
`
the content of a CommandBar is not shown.
Screenshot

### Expected behavior

The content is visible with the MaterialToolkitTheme.

### How to reproduce it (as minimally and precisely as possible)

[UnoApp1.zip](https://github.com/unoplatform/uno/files/15418952/UnoApp1.zip)

### Workaround

Comment ` <utum:MaterialToolkitTheme ColorOverrideSource="ms-appx:///Styles/ColorPaletteOverride.xaml"> ` in App.xaml

### Works on UWP/WinUI

None

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

5.2.132

### Affected platforms

WebAssembly, Windows (WinAppSDK)

### IDE

Visual Studio 2022

### IDE version

17.10.0

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Kunal22shah**
﻿GitHub Issue: 
closes #1418
closes #1425 

## PR Type

What kind of change does this PR introduce?

- Bugfix
- Feature
- Code style update (formatting)
- Refactoring (no functional changes, no api changes)
- Build or CI related changes
- Documentation content changes
- Other... Please describe:

## PR Checklist 
Please check if your PR fulfills the following requirements:

- Commits must be following the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] Tested the changes where applicable:
	- [ ] UWP
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/Uno.Themes/tree/master/doc)
	- [ ] [material-controls-styles.md](https://github.com/unoplatform/Uno.Themes/blob/master/doc/material-controls-styles.md)
	- [ ] [cupertino-controls-styles.md](https://github.com/unoplatform/Uno.Themes/blob/master/doc/cupertino-controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/Uno.Themes/blob/master/doc/lightweight-styling.md)
- [ ] Contains **No** breaking changes
  > If the pull request contains breaking changes, commit message must contain a detailed description of the action to take for the consumer of this library. As explained by the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)

### CODE REVIEW

File: src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml [hunk 1]
```diff
@@ -114,55 +65,37 @@
...
 		<Setter Property="Content" Value="" />
 		<Setter Property="ContentTemplate">
 			<Setter.Value>
 				<DataTemplate>
-					<TextBlock Text="{Binding}"
-							   Style="{StaticResource MaterialTitleLarge}" />
+					<TextBlock Text="{Binding}" Style="{StaticResource MaterialTitleLarge}" />
 				</DataTemplate>
 			</Setter.Value>
 		</Setter>
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:78]: I think we can remove these now, we should add a CommandBar sample page if we don't already have one and test what this looks like without these Setters

File: src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml [hunk 2]
```diff
@@ -24,86 +25,49 @@
...
-			  win:Height="48"
-			  Background="{TemplateBinding Background}">
-
-			<Grid x:Name="ContentRoot"
-				  Margin="{TemplateBinding Padding}"
-				  Background="{TemplateBinding Background}"
-				  Opacity="{TemplateBinding Opacity}">
-
+	<!-- Define the XAML-based CommandBar Template -->
+	<ControlTemplate x:Key="MaterialXamlCommandBarTemplate" TargetType="CommandBar">
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:41]: We shouldn't need this template anymore. We can remove this ControlTemplate

File: src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml [hunk 3]
```diff
@@ -114,55 +78,30 @@
...
 		<Setter Property="HorizontalAlignment" Value="Stretch" />
 		<Setter Property="VerticalAlignment" Value="Top" />
 		<Setter Property="HorizontalContentAlignment" Value="Left" />
 		<Setter Property="VerticalContentAlignment" Value="Center" />
-
 		<Setter Property="OverflowButtonVisibility" Value="Collapsed" />
 		<Setter Property="IsDynamicOverflowEnabled" Value="False" />
-		
 		<!-- Workaround for WinUI issue: https://github.com/microsoft/microsoft-ui-xaml/issues/6388 -->
 		<Setter Property="toolkit:CommandBarExtensions.NavigationCommand" Value="{x:Null}" />
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:95]: Remove

Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:95]: Remove

File: src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml
Comment (agneszitte) [inline src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:1]: @kazo0, @Xiaoy312 do we need any additional doc adjustments as well?
A note in https://github.com/unoplatform/Uno.Themes/blob/master/doc/material-migration.md ?

Comment (Xiaoy312) [reply src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:1]: we dont need any migration note here
in fact, we probably needed one with native m3
but not anymore with these styles gone

Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:1]: please install&/configure XamlStyler and format the file with that
you are introducing a lot of necessary whitespace/newline changes for no goods here

**Review by Xiaoy312 — APPROVED** [review]

File: src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml [hunk 4]
```diff
@@ -24,145 +25,28 @@
...
-								 MaxWidth="{ThemeResource AppBarButtonContentHeight}"
-								 Margin="0,12"
-								 HorizontalAlignment="Stretch"
-								 StretchDirection="DownOnly"
-								 AutomationProperties.AccessibilityView="Raw">
-							<ContentPresenter x:Name="Content"
-											  Content="{TemplateBinding Icon}"
-											  Foreground="{TemplateBinding Foreground}" />
+					<um:Ripple Feedback="{TemplateBinding Foreground}" FeedbackOpacity="{StaticResource PressedOpacity}">
+						<Viewbox x:Name="ContentViewbox" MaxHeight="{ThemeResource AppBarButtonContentHeight}" MaxWidth="{ThemeResource AppBarButtonContentHeight}" Margin="0,12" HorizontalAlignment="Stretch" StretchDirection="DownOnly" AutomationProperties.AccessibilityView="Raw">
```
Comment (Xiaoy312) [inline src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:29]: image

Comment (Kunal22shah) [reply src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:29]: thanks, i've updated my settings

**Comment by kazo0** [comment]
@Kunal22shah we should edit the CommandBarSamplePage.xaml to no longer navigate to a separate stack of pages and just display a normal CommandBar directly in CommandBarSamplePage.xaml. Similar to how other sample pages are done like Button and CheckBox.

You can see we currently direct people to click a Button to go to a new page where we have a CommandBar at the top and some text talking about the back button on the CommandBar. All of that isn't relevant anymore as the CommandBar is no longer being used for navigation as it was before:

image

Let me know if you want to pair to work through as I know this one is a bit messy

File: src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml (hunk 4 above)
Comment (Xiaoy312) [reply src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml]: please take a look at the diff set
your xamlstyler is not setup as expected

**Comment by Kunal22shah** [comment]
> @Kunal22shah we should edit the CommandBarSamplePage.xaml to no longer navigate to a separate stack of pages and just display a normal CommandBar directly in CommandBarSamplePage.xaml. Similar to how other sample pages are done like Button and CheckBox.
> 
> You can see we currently direct people to click a Button to go to a new page where we have a CommandBar at the top and some text talking about the back button on the CommandBar. All of that isn't relevant anymore as the CommandBar is no longer being used for navigation as it was before:
> 
> image
> 
> Let me know if you want to pair to work through as I know this one is a bit messy

@kazo0 Done!

File: src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml [hunk 5]
```diff
@@ -1,168 +1,72 @@
...
+                    </um:Ripple>
+                </ControlTemplate>
+            </Setter.Value>
+        </Setter>
+    </Style>
+    <!--  Define the base style for Material CommandBar  -->
+    <Style x:Key="MaterialBaseCommandBarStyle" TargetType="CommandBar">
+        <Setter Property="Background" Value="{ThemeResource SurfaceBrush}" />
+        <Setter Property="Foreground" Value="{ThemeResource OnSurfaceBrush}" />
+        <Setter Property="Height" Value="{StaticResource MaterialCommandBarHeight}" />
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml:52]: We can remove the hardcoded Height

File: src/samples/UWP/Uno.Themes.Samples.Shared/Content/Controls/CommandBarSamplePage.xaml [hunk 6]
```diff
@@ -6,44 +6,35 @@
...
+    <Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+        <sample:SamplePageLayout>
+            <sample:SamplePageLayout.M3MaterialTemplate>
+                <DataTemplate>
+                    <StackPanel>
+                        <Grid Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">
+                            <StackPanel>
+                                <smtx:XamlDisplay Margin="16" UniqueKey="CommandBarSamplePage">
+                                    <CommandBar
+                                        toolkit:VisibleBoundsPadding.PaddingMask="Top"
```
Comment (kazo0) [inline src/samples/UWP/Uno.Themes.Samples.Shared/Content/Controls/CommandBarSamplePage.xaml:23]: Shouldn't need this PaddingMask

**Review by kazo0 — APPROVED** [review]

File: src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml (hunk 4 above)
Comment (kazo0) [reply src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml]: image


Make sure you are using tabs/mixed as the indentation style. I see spaces being used here

Comment (Kunal22shah) [reply src/library/Uno.Material/Styles/Controls/v2/CommandBar.xaml]: yes, currently fixing my styler setting.

**Comment by kazo0** [comment]
@Mergifyio backport release/stable/5.1

**Pull request merged**